server/controllers/availability.js

```diff
@@ -0,0 +1,104 @@
+const Availability = require("../models/availability");
+const asyncHandler = require("express-async-handler");
+const Profile = require("../models/Profile");
+
+// @route POST /schedule/
+// @desc Create a schedule
+// @access Public
+exports.createSchedule = asyncHandler(async (req, res, next) => {
+  const profile = await User.findById(req.user.id, "profile");
+
+  if (!profile) {
+    res.status(401);
+    throw new Error("Not authorized");
+  }
+  const profileId = await Profile.findOne({ userId: req.user.id });
+  const id = profileId._id;
+  const { name, days } = req.body;
+  const schedule = await Availability.findOne({ name, petSitterId: profileId._id });
+  if (schedule) {
+    res.status(400);
+    throw new Error("Schedule name already exist");
+  }
+  const newSchedule = Availability.create({
+    petSitterId: id,
+    name,
+    days,
+  });
+  if (newSchedule) {
+    if (!profileId.activeSchedule) {
+      profileId.set({ activeSchedule: newSchedule.id });
+      await profileId.save();
+    }
+    res.status(201).json({
+      success: {
+        newSchedule
+      }
+    });
+  }
+});
+
+// @route GET /availability/:scheduleId
+// @desc Get schedule
+// @access Private
+exports.getScheduleId = asyncHandler(async (req, res, next) => {
+  const scheduleId = req.params.scheduleId;
+  const schedule = await Availability.findOne({ _id: scheduleId })
+  if (schedule) {
+    res.status(200);
+    res.send(schedule);
+  } else {
+    res.status(404).json({});
+    throw new Error("No Schedule found");
+  }
+});
+
+// @route GET /availability
+// @desc Get schedule
+// @access Private
+exports.getSchedule = asyncHandler(async (req, res, next) => {
+  const profileId = await Profile.findOne({ userId: req.user.id });
+  const id = profileId._id;
+  const schedule = await Availability.find({ petSitterId: id })
+  if (schedule) {
```

Comment:
We don't need this `if` clause as the mongoose query `Model.find()` returns an empty list if no matching docs are found (which is a truthy value)

Reply:
Removed the if clause.